js/list.js

```diff
@@ -0,0 +1,204 @@
+                newDiv.appendChild(dateTextNode);
+                document.getElementById("content").appendChild(newDiv);
+                i++;
+            }
+
+            //load the count value and write it on the page
+            totalRecord = arr.length;
+            console.log("Total Record : " + totalRecord);
+            document.getElementById("count-label").innerHTML = "Count: " + totalRecord;
+
+            setTimeout(function () {
+                //check for edits
+                const allCardsArray = document.getElementsByClassName("card");
+                for (var i = 0; i < allCardsArray.length; i++) {
+                    let card = allCardsArray[i];
+                    card.addEventListener('click', function () {
+                        let cardName = card.childNodes[0].nodeValue;
+                        let cardDescription = card.childNodes[2].nodeValue;
+                        let cardDate = card.childNodes[4].nodeValue;
+                        //console.log(cardName);console.log(cardDate);
+                        const c = arr.find(item => item.name == cardName);
+                        //console.log(c);
+                        //console.log(c.key);
+                        set(ref(db, 'eventToEdit/'),
+                            {
+                                id: c.key,
+                                name: cardName,
+                                description: cardDescription,
+                                date: cardDate
+                            });
+                        window.location.href = 'form.html';
+                    });
+                }
+            }, 2000);
+
+
+        } else {
+            console.log("No data available");
+        }
+    }).catch((error) => {
+        console.error(error);
+    });
+}
+
+//This function deletes all the card events in the page
+function clearEvents() {
+    var allCards = document.querySelectorAll('.card');
+    allCards.forEach(c => {
+        c.remove();
+    });
+}
+
+//This function converts firebase snapshot to array
+function snapshotToArray(snapshot) {
+    var returnArr = [];
+    snapshot.forEach(function (childSnapshot) {
+        var item = childSnapshot.val();
+        item.key = childSnapshot.key;
+        returnArr.push(item);
+    });
+    return returnArr;
+}
+
+//This function compares one date with the current date and returns 1 if it's a future date, 0 the current date, -1 a past date
+function compareDates(divDate) {
+    const currDate = new Date();
+    //console.log("Today date"+currDate.toLocaleDateString());
+    //console.log("div date"+divDate);
+
+    //current date format : 8/19/2023
+    let currDay = currDate.getDate();
+    let currMonth = currDate.getMonth() + 1;
+    let currYear = currDate.getFullYear();
+
+    //div date format : 2023-08-31T14:40   -  type="datetime-local" 
+    var divDay = divDate.substring(8, 10);
+    var divMonth = divDate.substring(5, 7);
+    var divYear = divDate.substring(0, 4);
```

Comment:
why not use date functions (getDate(), getMonth(), getFullYear()) ?

Reply:
done